// @angular-three/core/fesm2015/angular-three-core-group.mjs
import { __classPrivateFieldSet, __classPrivateFieldGet } from 'tslib';
import * as i1 from '@angular-three/core';
import { NGT_OBJECT_WATCHED_CONTROLLER, NGT_OBJECT_CONTROLLER_PROVIDER } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, Inject, NgModule } from '@angular/core';
import * as THREE from 'three';

var _NgtGroup_group;
class NgtGroup {
    constructor(objectController, ngZone) {
        this.objectController = objectController;
        this.ngZone = ngZone;
        _NgtGroup_group.set(this, void 0);
        objectController.initFn = () => {
            __classPrivateFieldSet(this, _NgtGroup_group, new THREE.Group(), "f");
            return __classPrivateFieldGet(this, _NgtGroup_group, "f");
        };
    }
    get group() {
        return __classPrivateFieldGet(this, _NgtGroup_group, "f");
    }
    ngOnInit() {
        this.ngZone.runOutsideAngular(() => {
            this.objectController.init();
        });
    }
}
_NgtGroup_group = new WeakMap();
NgtGroup.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtGroup, deps: [{ token: NGT_OBJECT_WATCHED_CONTROLLER }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive });
NgtGroup.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtGroup, selector: "ngt-group", providers: [NGT_OBJECT_CONTROLLER_PROVIDER], exportAs: ["ngtGroup"], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtGroup, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-group',
                    exportAs: 'ngtGroup',
                    providers: [NGT_OBJECT_CONTROLLER_PROVIDER],
                }]
        }], ctorParameters: function () {
        return [{ type: i1.NgtObject3dController, decorators: [{
                        type: Inject,
                        args: [NGT_OBJECT_WATCHED_CONTROLLER]
                    }] }, { type: i0.NgZone }];
    } });
class NgtGroupModule {
}
NgtGroupModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtGroupModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtGroupModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtGroupModule, declarations: [NgtGroup], exports: [NgtGroup] });
NgtGroupModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtGroupModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtGroupModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtGroup],
                    exports: [NgtGroup],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { NgtGroup, NgtGroupModule };
