// ngx-three/esm2020/lib/stats/stats.directive.mjs
import { Directive, Input } from '@angular/core';
import Stats from 'three/examples/jsm/libs/stats.module';
import * as i0 from "@angular/core";
import * as i1 from "../ThEngine.service";
export class StatsDirective {
    constructor(engineService) {
        this.engineService = engineService;
        this.stats = Stats();
    }
    ngOnDestroy() {
        if (this.renderSubscription) {
            this.renderSubscription.unsubscribe();
        }
    }
    ngOnInit() {
        if (!this.engineService.canvas) {
            throw new Error('No canvas present');
        }
        const parentElement = this.engineService.canvas.parentElement;
        if (parentElement) {
            parentElement.style.position = 'relative';
            this.stats.domElement.style.position = 'absolute';
            this.stats.showPanel(0);
            this.engineService.canvas.parentElement?.appendChild(this.stats.dom);
            this.renderSubscription = this.engineService.beforeRender$.subscribe(() => {
                this.stats.update();
            });
        }
    }
    set thStats(enabled) {
        if (!enabled) {
            this.stats.domElement.style.visibility = 'hidden';
        }
        else {
            this.stats.domElement.style.visibility = 'visible';
        }
    }
}
StatsDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: StatsDirective, deps: [{ token: i1.ThEngineService }], target: i0.ɵɵFactoryTarget.Directive });
StatsDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.2", type: StatsDirective, selector: "[thStats]", inputs: { thStats: "thStats" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: StatsDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[thStats]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThEngineService }]; }, propDecorators: { thStats: [{
                type: Input
            }] } });
