// ngx-three/esm2020/lib/pipes/vector.pipe.mjs
import { Pipe } from '@angular/core';
import { Vector2, Vector3, Vector4 } from 'three';
import * as i0 from "@angular/core";
/**
 * transform arrays to vectors
 */
export class Vector2Pipe {
    transform(args) {
        return new Vector2(...args);
    }
}
Vector2Pipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: Vector2Pipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
Vector2Pipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: Vector2Pipe, name: "vector2" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: Vector2Pipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'vector2',
                    pure: true
                }]
        }] });
export class Vector3Pipe {
    transform(args) {
        return new Vector3(...args);
    }
}
Vector3Pipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: Vector3Pipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
Vector3Pipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: Vector3Pipe, name: "vector3" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: Vector3Pipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'vector3',
                    pure: true
                }]
        }] });
export class Vector4Pipe {
    transform(args) {
        return new Vector4(...args);
    }
}
Vector4Pipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: Vector4Pipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
Vector4Pipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: Vector4Pipe, name: "vector4" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: Vector4Pipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'vector4',
                    pure: true
                }]
        }] });
