// ngx-three/esm2020/lib/events/raycaster.events.directive.mjs
import { Directive, EventEmitter, Host, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../generated/ThObject3D";
import * as i2 from "./raycaster.service";
export class RaycasterEventDirective {
    constructor(host, raycasterService) {
        this.host = host;
        this.raycasterService = raycasterService;
        this.mouseEnter = new EventEmitter();
        this.mouseExit = new EventEmitter();
        /* eslint-disable @angular-eslint/no-output-native */
        this.click = new EventEmitter();
        this.onMouseEnter = this.onMouseEnter.bind(this);
        this.onMouseExit = this.onMouseExit.bind(this);
        this.onClick = this.onClick.bind(this);
    }
    ngAfterViewInit() {
        this.subscribeEvents();
    }
    subscribeEvents() {
        this.raycasterService.addEventTarget(this);
        /*
        const obj = this.host.obj;
        // TODO: addEventListener, removeEventListener for proxy
        if (this.mouseEnter.observers.length > 0) {
          obj.addEventListener(RaycasterEvent.mouseEnter, this.onMouseEnter);
        }
        if (this.mouseExit.observers.length > 0) {
          obj.addEventListener(RaycasterEvent.mouseExit, this.onMouseExit);
        }
        if (this.click.observers.length > 0) {
          obj.addEventListener(RaycasterEvent.click, this.onClick);
        }
        */
    }
    unSubscribeEvents() {
        /*
        const obj = this.host.obj;
        if (obj) {
          obj.removeEventListener(RaycasterEvent.mouseEnter, this.onMouseEnter);
          obj.removeEventListener(RaycasterEvent.mouseExit, this.onMouseExit);
          obj.removeEventListener(RaycasterEvent.click, this.onClick);
        }
        */
        this.raycasterService.removeEventTarget(this);
    }
    onMouseExit() {
        this.mouseExit.emit({
            component: this.host
        });
    }
    onMouseEnter(event) {
        // console.log('RaycasterGroupDirective.onMouseEnter', event);
        this.mouseEnter.emit({
            component: this.host,
            face: event.face
        });
    }
    onClick(event) {
        // console.log('onClick', event);
        this.click.emit({
            component: this.host,
            face: event.face
        });
    }
    ngOnDestroy() {
        this.unSubscribeEvents();
    }
}
RaycasterEventDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: RaycasterEventDirective, deps: [{ token: i1.ThObject3D, host: true }, { token: i2.RaycasterService }], target: i0.ɵɵFactoryTarget.Directive });
RaycasterEventDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.2", type: RaycasterEventDirective, selector: "[onClick], [onMouseEnter], [onMouseExit]", outputs: { mouseEnter: "onMouseEnter", mouseExit: "onMouseExit", click: "onClick" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: RaycasterEventDirective, decorators: [{
            type: Directive,
            args: [{ selector: '[onClick], [onMouseEnter], [onMouseExit]' }]
        }], ctorParameters: function () { return [{ type: i1.ThObject3D, decorators: [{
                    type: Host
                }] }, { type: i2.RaycasterService }]; }, propDecorators: { mouseEnter: [{
                type: Output,
                args: ['onMouseEnter']
            }], mouseExit: [{
                type: Output,
                args: ['onMouseExit']
            }], click: [{
                type: Output,
                args: ['onClick']
            }] } });
