// ngx-three/esm2020/lib/renderer/th-render.directive.mjs
import { Directive, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../ThEngine.service";
export class ThRenderDirective {
    constructor(engineService) {
        this.engineService = engineService;
    }
    get beforeRender() {
        return this.engineService.beforeRender$;
    }
}
ThRenderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThRenderDirective, deps: [{ token: i1.ThEngineService }], target: i0.ɵɵFactoryTarget.Directive });
ThRenderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.2", type: ThRenderDirective, selector: "[beforeRender]", outputs: { beforeRender: "beforeRender" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThRenderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[beforeRender]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThEngineService }]; }, propDecorators: { beforeRender: [{
                type: Output
            }] } });
