// @angular-three/core/esm2020/lines/lib/line-loop/line-loop.directive.mjs
// GENERATED
import { NgtCommonLine, NGT_MATERIAL_GEOMETRY_CONTROLLER_PROVIDER, NGT_OBJECT_TYPE, } from '@angular-three/core';
import { NgModule, Directive } from '@angular/core';
import * as THREE from 'three';
import * as i0 from "@angular/core";
export class NgtLineLoop extends NgtCommonLine {
}
NgtLineLoop.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtLineLoop, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtLineLoop.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtLineLoop, selector: "ngt-line-loop", providers: [
        {
            provide: NgtCommonLine,
            useExisting: NgtLineLoop,
        },
        NGT_MATERIAL_GEOMETRY_CONTROLLER_PROVIDER,
        {
            provide: NGT_OBJECT_TYPE,
            useValue: THREE.LineLoop,
        },
    ], exportAs: ["ngtLineLoop"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtLineLoop, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-loop',
                    exportAs: 'ngtLineLoop',
                    providers: [
                        {
                            provide: NgtCommonLine,
                            useExisting: NgtLineLoop,
                        },
                        NGT_MATERIAL_GEOMETRY_CONTROLLER_PROVIDER,
                        {
                            provide: NGT_OBJECT_TYPE,
                            useValue: THREE.LineLoop,
                        },
                    ],
                }]
        }] });
export class NgtLineLoopModule {
}
NgtLineLoopModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtLineLoopModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtLineLoopModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtLineLoopModule, declarations: [NgtLineLoop], exports: [NgtLineLoop] });
NgtLineLoopModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtLineLoopModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtLineLoopModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtLineLoop],
                    exports: [NgtLineLoop],
                }]
        }] });
