// @angular-three/core/esm2020/helpers/lib/arrow-helper/arrow-helper.directive.mjs
// GENERATED
import { NgtHelper, NGT_OBJECT_CONTROLLER_PROVIDER } from '@angular-three/core';
import { NgModule, Directive, Input } from '@angular/core';
import * as THREE from 'three';
import * as i0 from "@angular/core";
export class NgtArrowHelper extends NgtHelper {
    constructor() {
        super(...arguments);
        this.helperType = THREE.ArrowHelper;
    }
    set args(v) {
        this.helperArgs = v;
    }
}
NgtArrowHelper.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtArrowHelper, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtArrowHelper.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtArrowHelper, selector: "ngt-arrow-helper", inputs: { args: "args" }, providers: [
        {
            provide: NgtHelper,
            useExisting: NgtArrowHelper,
        },
        NGT_OBJECT_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtArrowHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtArrowHelper, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-arrow-helper',
                    exportAs: 'ngtArrowHelper',
                    providers: [
                        {
                            provide: NgtHelper,
                            useExisting: NgtArrowHelper,
                        },
                        NGT_OBJECT_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
export class NgtArrowHelperModule {
}
NgtArrowHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtArrowHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtArrowHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtArrowHelperModule, declarations: [NgtArrowHelper], exports: [NgtArrowHelper] });
NgtArrowHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtArrowHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtArrowHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtArrowHelper],
                    exports: [NgtArrowHelper],
                }]
        }] });
