// @angular-three/core/esm2020/cameras/lib/camera/camera.directive.mjs
// GENERATED
import { NgtCommonCamera, NGT_OBJECT_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { NgModule, Directive, Input } from '@angular/core';
import * as THREE from 'three';
import * as i0 from "@angular/core";
export class NgtCamera extends NgtCommonCamera {
    constructor() {
        super(...arguments);
        this.cameraType = THREE.Camera;
    }
    set args(v) {
        this.cameraArgs = v;
    }
}
NgtCamera.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCamera, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtCamera.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtCamera, selector: "ngt-camera", inputs: { args: "args" }, providers: [
        {
            provide: NgtCommonCamera,
            useExisting: NgtCamera,
        },
        NGT_OBJECT_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCamera, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-camera',
                    exportAs: 'ngtCamera',
                    providers: [
                        {
                            provide: NgtCommonCamera,
                            useExisting: NgtCamera,
                        },
                        NGT_OBJECT_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
export class NgtCameraModule {
}
NgtCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCameraModule, declarations: [NgtCamera], exports: [NgtCamera] });
NgtCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtCamera],
                    exports: [NgtCamera],
                }]
        }] });
