// @angular-three/core/esm2020/lib/controllers/audio.controller.mjs
// GENERATED
import { Directive, Input, NgModule } from '@angular/core';
import { Controller, createControllerProviderFactory } from './controller';
import * as i0 from "@angular/core";
export class NgtAudioController extends Controller {
    constructor() {
        super(...arguments);
        this.buffer = null;
        this.detune = 0;
        this.loop = false;
        this.loopStart = 0;
        this.loopEnd = 0;
        this.offset = 0;
        this.duration = undefined;
        this.playbackRate = 1;
        this.isPlaying = false;
        this.hasPlaybackControl = true;
        this.sourceType = 'empty';
        this.source = null;
        this.filters = [];
    }
    get props() {
        return [
            'autoplay',
            'buffer',
            'detune',
            'loop',
            'loopStart',
            'loopEnd',
            'offset',
            'duration',
            'playbackRate',
            'isPlaying',
            'hasPlaybackControl',
            'sourceType',
            'source',
            'filters',
        ];
    }
    get controller() {
        return this.audioController;
    }
}
NgtAudioController.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtAudioController, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtAudioController.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtAudioController, selector: "\n    ngt-audio,\n    ngt-positional-audio,\n  ", inputs: { autoplay: "autoplay", buffer: "buffer", detune: "detune", loop: "loop", loopStart: "loopStart", loopEnd: "loopEnd", offset: "offset", duration: "duration", playbackRate: "playbackRate", isPlaying: "isPlaying", hasPlaybackControl: "hasPlaybackControl", sourceType: "sourceType", source: "source", filters: "filters", audioController: "audioController" }, exportAs: ["ngtAudioController"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtAudioController, decorators: [{
            type: Directive,
            args: [{
                    selector: `
    ngt-audio,
    ngt-positional-audio,
  `,
                    exportAs: 'ngtAudioController',
                }]
        }], propDecorators: { autoplay: [{
                type: Input
            }], buffer: [{
                type: Input
            }], detune: [{
                type: Input
            }], loop: [{
                type: Input
            }], loopStart: [{
                type: Input
            }], loopEnd: [{
                type: Input
            }], offset: [{
                type: Input
            }], duration: [{
                type: Input
            }], playbackRate: [{
                type: Input
            }], isPlaying: [{
                type: Input
            }], hasPlaybackControl: [{
                type: Input
            }], sourceType: [{
                type: Input
            }], source: [{
                type: Input
            }], filters: [{
                type: Input
            }], audioController: [{
                type: Input
            }] } });
export class NgtAudioControllerModule {
}
NgtAudioControllerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtAudioControllerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtAudioControllerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtAudioControllerModule, declarations: [NgtAudioController], exports: [NgtAudioController] });
NgtAudioControllerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtAudioControllerModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtAudioControllerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtAudioController],
                    exports: [NgtAudioController],
                }]
        }] });
export const [NGT_AUDIO_WATCHED_CONTROLLER, NGT_AUDIO_CONTROLLER_PROVIDER] = createControllerProviderFactory({
    watchedControllerTokenName: 'Watched AudioController',
    controller: NgtAudioController,
});
