// @angular-three/core/esm2020/lib/controllers/controller.mjs
import { ChangeDetectorRef, Directive, InjectionToken, Input, NgZone, Optional, } from '@angular/core';
import { ReplaySubject, takeUntil } from 'rxjs';
import { NgtDestroyedService } from '../services/destroyed.service';
import * as i0 from "@angular/core";
export class Controller {
    constructor(ngZone) {
        this.ngZone = ngZone;
        this.disabled = false;
        this.change$ = new ReplaySubject(1);
    }
    ngOnChanges(changes) {
        if (this.controller) {
            this.controller.ngOnChanges(changes);
            this.change$.next(changes);
        }
        else {
            this.change$.next(changes);
        }
    }
    ngOnInit() {
        if (this.controller) {
            this.ngZone.runOutsideAngular(() => {
                this.props.forEach((prop) => {
                    this[prop] = this.controller[prop];
                });
            });
        }
    }
}
Controller.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: Controller, deps: [{ token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive });
Controller.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: Controller, inputs: { disabled: "disabled" }, usesOnChanges: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: Controller, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i0.NgZone }]; }, propDecorators: { disabled: [{
                type: Input
            }] } });
export function controllerFactory(newInstanceOnNull = false, controllerType) {
    return (controller, changeDetectorRef, destroyed) => {
        if (!controller) {
            return newInstanceOnNull ? new controllerType() : null;
        }
        controller.change$.pipe(takeUntil(destroyed)).subscribe(() => {
            changeDetectorRef.markForCheck();
        });
        return controller;
    };
}
export function createControllerProviderFactory({ watchedControllerTokenName, controller, newInstanceOnNull = false, }) {
    const watchedControllerToken = new InjectionToken(watchedControllerTokenName);
    const controllerProvider = [
        NgtDestroyedService,
        {
            provide: watchedControllerToken,
            deps: [
                [new Optional(), controller],
                ChangeDetectorRef,
                NgtDestroyedService,
            ],
            useFactory: controllerFactory(newInstanceOnNull, controller),
        },
    ];
    return [watchedControllerToken, controllerProvider];
}
