// @angular-three/core/esm2020/cameras/lib/stereo-camera/stereo-camera.directive.mjs
// GENERATED
import { NgtCommonCamera, NGT_OBJECT_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { NgModule, Directive, Input } from '@angular/core';
import * as THREE from 'three';
import * as i0 from "@angular/core";
export class NgtStereoCamera extends NgtCommonCamera {
    constructor() {
        super(...arguments);
        this.cameraType = THREE.StereoCamera;
    }
    set args(v) {
        this.cameraArgs = v;
    }
}
NgtStereoCamera.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtStereoCamera, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtStereoCamera.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtStereoCamera, selector: "ngt-stereo-camera", inputs: { args: "args" }, providers: [
        {
            provide: NgtCommonCamera,
            useExisting: NgtStereoCamera,
        },
        NGT_OBJECT_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtStereoCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtStereoCamera, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-stereo-camera',
                    exportAs: 'ngtStereoCamera',
                    providers: [
                        {
                            provide: NgtCommonCamera,
                            useExisting: NgtStereoCamera,
                        },
                        NGT_OBJECT_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
export class NgtStereoCameraModule {
}
NgtStereoCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtStereoCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtStereoCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtStereoCameraModule, declarations: [NgtStereoCamera], exports: [NgtStereoCamera] });
NgtStereoCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtStereoCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtStereoCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtStereoCamera],
                    exports: [NgtStereoCamera],
                }]
        }] });
