// @angular-three/core/esm2020/lights/lib/light-probe/light-probe.directive.mjs
// GENERATED
import { NgtLight, NGT_OBJECT_CONTROLLER_PROVIDER } from '@angular-three/core';
import { NgModule, Directive, Input } from '@angular/core';
import * as THREE from 'three';
import * as i0 from "@angular/core";
export class NgtLightProbe extends NgtLight {
    constructor() {
        super(...arguments);
        this.lightType = THREE.LightProbe;
    }
    set args(v) {
        this.lightArgs = v;
    }
}
NgtLightProbe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtLightProbe, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtLightProbe.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtLightProbe, selector: "ngt-light-probe", inputs: { args: "args" }, providers: [
        {
            provide: NgtLight,
            useExisting: NgtLightProbe,
        },
        NGT_OBJECT_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtLightProbe"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtLightProbe, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-light-probe',
                    exportAs: 'ngtLightProbe',
                    providers: [
                        {
                            provide: NgtLight,
                            useExisting: NgtLightProbe,
                        },
                        NGT_OBJECT_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
export class NgtLightProbeModule {
}
NgtLightProbeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtLightProbeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtLightProbeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtLightProbeModule, declarations: [NgtLightProbe], exports: [NgtLightProbe] });
NgtLightProbeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtLightProbeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtLightProbeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtLightProbe],
                    exports: [NgtLightProbe],
                }]
        }] });
