// @angular-three/core/esm2020/lib/three/sprite.mjs
var _NgtCommonSprite_sprite;
import { __classPrivateFieldGet, __classPrivateFieldSet } from "tslib";
import { ContentChild, Directive, Inject, Input, NgZone, } from '@angular/core';
import * as THREE from 'three';
import { NGT_OBJECT_CONTROLLER_PROVIDER, NgtObject3dController, } from '../controllers/object-3d.controller';
import { NgtMaterial } from './material';
import * as i0 from "@angular/core";
import * as i1 from "../controllers/object-3d.controller";
export class NgtCommonSprite {
    constructor(objectController, ngZone) {
        this.objectController = objectController;
        this.ngZone = ngZone;
        _NgtCommonSprite_sprite.set(this, void 0);
        objectController.initFn = () => {
            if (this.material) {
                __classPrivateFieldSet(this, _NgtCommonSprite_sprite, new this.spriteType(this.material), "f");
            }
            else if (this.materialDirective) {
                if (this.materialDirective.material instanceof THREE.SpriteMaterial) {
                    __classPrivateFieldSet(this, _NgtCommonSprite_sprite, new this.spriteType(this.materialDirective.material), "f");
                }
            }
            return __classPrivateFieldGet(this, _NgtCommonSprite_sprite, "f");
        };
    }
    ngOnInit() {
        this.ngZone.runOutsideAngular(() => {
            this.objectController.init();
        });
    }
    get sprite() {
        return __classPrivateFieldGet(this, _NgtCommonSprite_sprite, "f");
    }
}
_NgtCommonSprite_sprite = new WeakMap();
NgtCommonSprite.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCommonSprite, deps: [{ token: NGT_OBJECT_CONTROLLER_PROVIDER }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive });
NgtCommonSprite.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtCommonSprite, inputs: { material: "material" }, queries: [{ propertyName: "materialDirective", first: true, predicate: NgtMaterial, descendants: true }], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCommonSprite, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i1.NgtObject3dController, decorators: [{
                    type: Inject,
                    args: [NGT_OBJECT_CONTROLLER_PROVIDER]
                }] }, { type: i0.NgZone }]; }, propDecorators: { material: [{
                type: Input
            }], materialDirective: [{
                type: ContentChild,
                args: [NgtMaterial]
            }] } });
