// ngx-three/esm2020/lib/generated/ThAdaptiveToneMappingPass.mjs
/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { AdaptiveToneMappingPass } from 'three/examples/jsm/postprocessing/AdaptiveToneMappingPass';
import { ThPassBase } from '../ThPassBase';
import { ThPass } from './ThPass';
import * as i0 from "@angular/core";
export class ThAdaptiveToneMappingPass extends ThPass {
    getType() {
        return AdaptiveToneMappingPass;
    }
    set adaptive(value) {
        if (this._objRef) {
            this._objRef.adaptive = value;
        }
    }
    set resolution(value) {
        if (this._objRef) {
            this._objRef.resolution = value;
        }
    }
    set needsInit(value) {
        if (this._objRef) {
            this._objRef.needsInit = value;
        }
    }
    set luminanceRT(value) {
        if (this._objRef) {
            this._objRef.luminanceRT = value;
        }
    }
    set previousLuminanceRT(value) {
        if (this._objRef) {
            this._objRef.previousLuminanceRT = value;
        }
    }
    set currentLuminanceRT(value) {
        if (this._objRef) {
            this._objRef.currentLuminanceRT = value;
        }
    }
    set copyUniforms(value) {
        if (this._objRef) {
            this._objRef.copyUniforms = value;
        }
    }
    set materialCopy(value) {
        if (this._objRef) {
            this._objRef.materialCopy = value;
        }
    }
    set materialLuminance(value) {
        if (this._objRef) {
            this._objRef.materialLuminance = value;
        }
    }
    set adaptLuminanceShader(value) {
        if (this._objRef) {
            this._objRef.adaptLuminanceShader = value;
        }
    }
    set materialAdaptiveLum(value) {
        if (this._objRef) {
            this._objRef.materialAdaptiveLum = value;
        }
    }
    set materialToneMap(value) {
        if (this._objRef) {
            this._objRef.materialToneMap = value;
        }
    }
    set fsQuad(value) {
        if (this._objRef) {
            this._objRef.fsQuad = value;
        }
    }
}
ThAdaptiveToneMappingPass.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThAdaptiveToneMappingPass, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThAdaptiveToneMappingPass.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThAdaptiveToneMappingPass, selector: "th-adaptiveToneMappingPass", inputs: { adaptive: "adaptive", resolution: "resolution", needsInit: "needsInit", luminanceRT: "luminanceRT", previousLuminanceRT: "previousLuminanceRT", currentLuminanceRT: "currentLuminanceRT", copyUniforms: "copyUniforms", materialCopy: "materialCopy", materialLuminance: "materialLuminance", adaptLuminanceShader: "adaptLuminanceShader", materialAdaptiveLum: "materialAdaptiveLum", materialToneMap: "materialToneMap", fsQuad: "fsQuad" }, providers: [
        {
            provide: ThPassBase,
            useExisting: forwardRef(() => ThAdaptiveToneMappingPass),
        },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThAdaptiveToneMappingPass, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-adaptiveToneMappingPass',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThPassBase,
                            useExisting: forwardRef(() => ThAdaptiveToneMappingPass),
                        },
                    ],
                }]
        }], propDecorators: { adaptive: [{
                type: Input
            }], resolution: [{
                type: Input
            }], needsInit: [{
                type: Input
            }], luminanceRT: [{
                type: Input
            }], previousLuminanceRT: [{
                type: Input
            }], currentLuminanceRT: [{
                type: Input
            }], copyUniforms: [{
                type: Input
            }], materialCopy: [{
                type: Input
            }], materialLuminance: [{
                type: Input
            }], adaptLuminanceShader: [{
                type: Input
            }], materialAdaptiveLum: [{
                type: Input
            }], materialToneMap: [{
                type: Input
            }], fsQuad: [{
                type: Input
            }] } });
