// ngx-three/esm2020/lib/generated/ThAmbientLightProbe.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, } from '@angular/core';
import { AmbientLightProbe } from 'three';
import { ThLightProbe } from './ThLightProbe';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThAmbientLightProbe extends ThLightProbe {
    getType() {
        return AmbientLightProbe;
    }
}
ThAmbientLightProbe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThAmbientLightProbe, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThAmbientLightProbe.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThAmbientLightProbe, selector: "th-ambientLightProbe", providers: [
        { provide: ThObject3D, useExisting: forwardRef(() => ThAmbientLightProbe) },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThAmbientLightProbe, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-ambientLightProbe',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        { provide: ThObject3D, useExisting: forwardRef(() => ThAmbientLightProbe) },
                    ],
                }]
        }] });
