// ngx-three/esm2020/lib/generated/ThBokehPass.mjs
/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { BokehPass, } from 'three/examples/jsm/postprocessing/BokehPass';
import { ThPassBase } from '../ThPassBase';
import { applyValue } from '../util';
import { ThPass } from './ThPass';
import * as i0 from "@angular/core";
export class ThBokehPass extends ThPass {
    getType() {
        return BokehPass;
    }
    set scene(value) {
        if (this._objRef) {
            this._objRef.scene = value;
        }
    }
    set camera(value) {
        if (this._objRef) {
            this._objRef.camera = value;
        }
    }
    set renderTargetColor(value) {
        if (this._objRef) {
            this._objRef.renderTargetColor = value;
        }
    }
    set renderTargetDepth(value) {
        if (this._objRef) {
            this._objRef.renderTargetDepth = value;
        }
    }
    set materialDepth(value) {
        if (this._objRef) {
            this._objRef.materialDepth = value;
        }
    }
    set materialBokeh(value) {
        if (this._objRef) {
            this._objRef.materialBokeh = value;
        }
    }
    set uniforms(map) {
        if (this._objRef) {
            Object.assign(this._objRef.uniforms, map);
        }
    }
    set fsQuad(value) {
        if (this._objRef) {
            this._objRef.fsQuad = value;
        }
    }
    set oldClearColor(value) {
        if (this._objRef) {
            this._objRef.oldClearColor = applyValue(this._objRef.oldClearColor, value);
        }
    }
}
ThBokehPass.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThBokehPass, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThBokehPass.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThBokehPass, selector: "th-bokehPass", inputs: { scene: "scene", camera: "camera", renderTargetColor: "renderTargetColor", renderTargetDepth: "renderTargetDepth", materialDepth: "materialDepth", materialBokeh: "materialBokeh", uniforms: "uniforms", fsQuad: "fsQuad", oldClearColor: "oldClearColor" }, providers: [
        { provide: ThPassBase, useExisting: forwardRef(() => ThBokehPass) },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThBokehPass, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-bokehPass',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        { provide: ThPassBase, useExisting: forwardRef(() => ThBokehPass) },
                    ],
                }]
        }], propDecorators: { scene: [{
                type: Input
            }], camera: [{
                type: Input
            }], renderTargetColor: [{
                type: Input
            }], renderTargetDepth: [{
                type: Input
            }], materialDepth: [{
                type: Input
            }], materialBokeh: [{
                type: Input
            }], uniforms: [{
                type: Input
            }], fsQuad: [{
                type: Input
            }], oldClearColor: [{
                type: Input
            }] } });
