// ngx-three/esm2020/lib/generated/ThCameraHelper.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { CameraHelper } from 'three';
import { ThLineSegments } from './ThLineSegments';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThCameraHelper extends ThLineSegments {
    getType() {
        return CameraHelper;
    }
    set camera(value) {
        if (this._objRef) {
            this._objRef.camera = value;
        }
    }
    set pointMap(value) {
        if (this._objRef) {
            this._objRef.pointMap = value;
        }
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
}
ThCameraHelper.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThCameraHelper, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThCameraHelper.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThCameraHelper, selector: "th-cameraHelper", inputs: { camera: "camera", pointMap: "pointMap", type: "type" }, providers: [
        { provide: ThObject3D, useExisting: forwardRef(() => ThCameraHelper) },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThCameraHelper, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-cameraHelper',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        { provide: ThObject3D, useExisting: forwardRef(() => ThCameraHelper) },
                    ],
                }]
        }], propDecorators: { camera: [{
                type: Input
            }], pointMap: [{
                type: Input
            }], type: [{
                type: Input
            }] } });
