// ngx-three/esm2020/lib/generated/ThClearMaskPass.mjs
/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, } from '@angular/core';
import { ClearMaskPass } from 'three/examples/jsm/postprocessing/MaskPass';
import { ThPassBase } from '../ThPassBase';
import { ThPass } from './ThPass';
import * as i0 from "@angular/core";
export class ThClearMaskPass extends ThPass {
    getType() {
        return ClearMaskPass;
    }
}
ThClearMaskPass.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThClearMaskPass, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThClearMaskPass.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThClearMaskPass, selector: "th-clearMaskPass", providers: [
        { provide: ThPassBase, useExisting: forwardRef(() => ThClearMaskPass) },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThClearMaskPass, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-clearMaskPass',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        { provide: ThPassBase, useExisting: forwardRef(() => ThClearMaskPass) },
                    ],
                }]
        }] });
