// ngx-three/esm2020/lib/generated/ThDirectionalLightHelper.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { DirectionalLightHelper, } from 'three';
import { applyValue } from '../util';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThDirectionalLightHelper extends ThObject3D {
    getType() {
        return DirectionalLightHelper;
    }
    set light(value) {
        if (this._objRef) {
            this._objRef.light = value;
        }
    }
    set lightPlane(value) {
        if (this._objRef) {
            this._objRef.lightPlane = value;
        }
    }
    set targetLine(value) {
        if (this._objRef) {
            this._objRef.targetLine = value;
        }
    }
    set color(value) {
        if (this._objRef) {
            this._objRef.color = value;
        }
    }
    set matrix(value) {
        if (this._objRef) {
            this._objRef.matrix = applyValue(this._objRef.matrix, value);
        }
    }
    set matrixAutoUpdate(value) {
        if (this._objRef) {
            this._objRef.matrixAutoUpdate = value;
        }
    }
}
ThDirectionalLightHelper.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThDirectionalLightHelper, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThDirectionalLightHelper.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThDirectionalLightHelper, selector: "th-directionalLightHelper", inputs: { light: "light", lightPlane: "lightPlane", targetLine: "targetLine", color: "color", matrix: "matrix", matrixAutoUpdate: "matrixAutoUpdate" }, providers: [
        {
            provide: ThObject3D,
            useExisting: forwardRef(() => ThDirectionalLightHelper),
        },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThDirectionalLightHelper, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-directionalLightHelper',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThObject3D,
                            useExisting: forwardRef(() => ThDirectionalLightHelper),
                        },
                    ],
                }]
        }], propDecorators: { light: [{
                type: Input
            }], lightPlane: [{
                type: Input
            }], targetLine: [{
                type: Input
            }], color: [{
                type: Input
            }], matrix: [{
                type: Input
            }], matrixAutoUpdate: [{
                type: Input
            }] } });
