// ngx-three/esm2020/lib/generated/ThInstancedBufferGeometry.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { InstancedBufferGeometry } from 'three';
import { ThBufferGeometry } from './ThBufferGeometry';
import * as i0 from "@angular/core";
export class ThInstancedBufferGeometry extends ThBufferGeometry {
    getType() {
        return InstancedBufferGeometry;
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
    set isInstancedBufferGeometry(value) {
        if (this._objRef) {
            this._objRef.isInstancedBufferGeometry = value;
        }
    }
    set groups(value) {
        if (this._objRef) {
            this._objRef.groups = value;
        }
    }
    set instanceCount(value) {
        if (this._objRef) {
            this._objRef.instanceCount = value;
        }
    }
}
ThInstancedBufferGeometry.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThInstancedBufferGeometry, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThInstancedBufferGeometry.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThInstancedBufferGeometry, selector: "th-instancedBufferGeometry", inputs: { type: "type", isInstancedBufferGeometry: "isInstancedBufferGeometry", groups: "groups", instanceCount: "instanceCount" }, providers: [
        {
            provide: ThBufferGeometry,
            useExisting: forwardRef(() => ThInstancedBufferGeometry),
        },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThInstancedBufferGeometry, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-instancedBufferGeometry',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThBufferGeometry,
                            useExisting: forwardRef(() => ThInstancedBufferGeometry),
                        },
                    ],
                }]
        }], propDecorators: { type: [{
                type: Input
            }], isInstancedBufferGeometry: [{
                type: Input
            }], groups: [{
                type: Input
            }], instanceCount: [{
                type: Input
            }] } });
