// ngx-three/esm2020/lib/generated/ThLineBasicMaterial.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { LineBasicMaterial, } from 'three';
import { applyValue } from '../util';
import { ThMaterial } from './ThMaterial';
import * as i0 from "@angular/core";
export class ThLineBasicMaterial extends ThMaterial {
    getType() {
        return LineBasicMaterial;
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
    set color(value) {
        if (this._objRef) {
            this._objRef.color = applyValue(this._objRef.color, value);
        }
    }
    set linewidth(value) {
        if (this._objRef) {
            this._objRef.linewidth = value;
        }
    }
    set linecap(value) {
        if (this._objRef) {
            this._objRef.linecap = value;
        }
    }
    set linejoin(value) {
        if (this._objRef) {
            this._objRef.linejoin = value;
        }
    }
}
ThLineBasicMaterial.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThLineBasicMaterial, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThLineBasicMaterial.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThLineBasicMaterial, selector: "th-lineBasicMaterial", inputs: { type: "type", color: "color", linewidth: "linewidth", linecap: "linecap", linejoin: "linejoin" }, providers: [
        { provide: ThMaterial, useExisting: forwardRef(() => ThLineBasicMaterial) },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThLineBasicMaterial, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-lineBasicMaterial',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        { provide: ThMaterial, useExisting: forwardRef(() => ThLineBasicMaterial) },
                    ],
                }]
        }], propDecorators: { type: [{
                type: Input
            }], color: [{
                type: Input
            }], linewidth: [{
                type: Input
            }], linecap: [{
                type: Input
            }], linejoin: [{
                type: Input
            }] } });
