// ngx-three/esm2020/lib/generated/ThLightProbe.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { LightProbe } from 'three';
import { applyValue } from '../util';
import { ThLight } from './ThLight';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThLightProbe extends ThLight {
    getType() {
        return LightProbe;
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
    set sh(value) {
        if (this._objRef) {
            this._objRef.sh = applyValue(this._objRef.sh, value);
        }
    }
}
ThLightProbe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThLightProbe, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThLightProbe.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThLightProbe, selector: "th-lightProbe", inputs: { type: "type", sh: "sh" }, providers: [
        { provide: ThObject3D, useExisting: forwardRef(() => ThLightProbe) },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThLightProbe, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-lightProbe',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        { provide: ThObject3D, useExisting: forwardRef(() => ThLightProbe) },
                    ],
                }]
        }], propDecorators: { type: [{
                type: Input
            }], sh: [{
                type: Input
            }] } });
