// ngx-three/esm2020/lib/generated/ThLight.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { Light } from 'three';
import { applyValue } from '../util';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThLight extends ThObject3D {
    getType() {
        return Light;
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
    set color(value) {
        if (this._objRef) {
            this._objRef.color = applyValue(this._objRef.color, value);
        }
    }
    set intensity(value) {
        if (this._objRef) {
            this._objRef.intensity = value;
        }
    }
    set shadow(value) {
        if (this._objRef) {
            this._objRef.shadow = value;
        }
    }
    set shadowCameraFov(value) {
        if (this._objRef) {
            this._objRef.shadowCameraFov = value;
        }
    }
    set shadowCameraLeft(value) {
        if (this._objRef) {
            this._objRef.shadowCameraLeft = value;
        }
    }
    set shadowCameraRight(value) {
        if (this._objRef) {
            this._objRef.shadowCameraRight = value;
        }
    }
    set shadowCameraTop(value) {
        if (this._objRef) {
            this._objRef.shadowCameraTop = value;
        }
    }
    set shadowCameraBottom(value) {
        if (this._objRef) {
            this._objRef.shadowCameraBottom = value;
        }
    }
    set shadowCameraNear(value) {
        if (this._objRef) {
            this._objRef.shadowCameraNear = value;
        }
    }
    set shadowCameraFar(value) {
        if (this._objRef) {
            this._objRef.shadowCameraFar = value;
        }
    }
    set shadowBias(value) {
        if (this._objRef) {
            this._objRef.shadowBias = value;
        }
    }
    set shadowMapWidth(value) {
        if (this._objRef) {
            this._objRef.shadowMapWidth = value;
        }
    }
    set shadowMapHeight(value) {
        if (this._objRef) {
            this._objRef.shadowMapHeight = value;
        }
    }
}
ThLight.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThLight, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThLight.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThLight, selector: "th-light", inputs: { type: "type", color: "color", intensity: "intensity", shadow: "shadow", shadowCameraFov: "shadowCameraFov", shadowCameraLeft: "shadowCameraLeft", shadowCameraRight: "shadowCameraRight", shadowCameraTop: "shadowCameraTop", shadowCameraBottom: "shadowCameraBottom", shadowCameraNear: "shadowCameraNear", shadowCameraFar: "shadowCameraFar", shadowBias: "shadowBias", shadowMapWidth: "shadowMapWidth", shadowMapHeight: "shadowMapHeight" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThLight) }], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThLight, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-light',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThLight) }],
                }]
        }], propDecorators: { type: [{
                type: Input
            }], color: [{
                type: Input
            }], intensity: [{
                type: Input
            }], shadow: [{
                type: Input
            }], shadowCameraFov: [{
                type: Input
            }], shadowCameraLeft: [{
                type: Input
            }], shadowCameraRight: [{
                type: Input
            }], shadowCameraTop: [{
                type: Input
            }], shadowCameraBottom: [{
                type: Input
            }], shadowCameraNear: [{
                type: Input
            }], shadowCameraFar: [{
                type: Input
            }], shadowBias: [{
                type: Input
            }], shadowMapWidth: [{
                type: Input
            }], shadowMapHeight: [{
                type: Input
            }] } });
