// ngx-three/esm2020/lib/generated/ThLOD.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { LOD } from 'three';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThLOD extends ThObject3D {
    getType() {
        return LOD;
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
    set levels(value) {
        if (this._objRef) {
            this._objRef.levels = value;
        }
    }
    set autoUpdate(value) {
        if (this._objRef) {
            this._objRef.autoUpdate = value;
        }
    }
    set objects(value) {
        if (this._objRef) {
            this._objRef.objects = value;
        }
    }
}
ThLOD.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThLOD, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThLOD.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThLOD, selector: "th-lOD", inputs: { type: "type", levels: "levels", autoUpdate: "autoUpdate", objects: "objects" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThLOD) }], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThLOD, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-lOD',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThLOD) }],
                }]
        }], propDecorators: { type: [{
                type: Input
            }], levels: [{
                type: Input
            }], autoUpdate: [{
                type: Input
            }], objects: [{
                type: Input
            }] } });
