// ngx-three/esm2020/lib/generated/ThOrthographicCamera.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { OrthographicCamera } from 'three';
import { ThCamera } from './ThCamera';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThOrthographicCamera extends ThCamera {
    getType() {
        return OrthographicCamera;
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
    set zoom(value) {
        if (this._objRef) {
            this._objRef.zoom = value;
        }
    }
    set view(value) {
        if (this._objRef) {
            this._objRef.view = value;
        }
    }
    set left(value) {
        if (this._objRef) {
            this._objRef.left = value;
        }
    }
    set right(value) {
        if (this._objRef) {
            this._objRef.right = value;
        }
    }
    set top(value) {
        if (this._objRef) {
            this._objRef.top = value;
        }
    }
    set bottom(value) {
        if (this._objRef) {
            this._objRef.bottom = value;
        }
    }
    set near(value) {
        if (this._objRef) {
            this._objRef.near = value;
        }
    }
    set far(value) {
        if (this._objRef) {
            this._objRef.far = value;
        }
    }
}
ThOrthographicCamera.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThOrthographicCamera, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThOrthographicCamera.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThOrthographicCamera, selector: "th-orthographicCamera", inputs: { type: "type", zoom: "zoom", view: "view", left: "left", right: "right", top: "top", bottom: "bottom", near: "near", far: "far" }, providers: [
        {
            provide: ThObject3D,
            useExisting: forwardRef(() => ThOrthographicCamera),
        },
        { provide: ThCamera, useExisting: forwardRef(() => ThOrthographicCamera) },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThOrthographicCamera, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-orthographicCamera',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThObject3D,
                            useExisting: forwardRef(() => ThOrthographicCamera),
                        },
                        { provide: ThCamera, useExisting: forwardRef(() => ThOrthographicCamera) },
                    ],
                }]
        }], propDecorators: { type: [{
                type: Input
            }], zoom: [{
                type: Input
            }], view: [{
                type: Input
            }], left: [{
                type: Input
            }], right: [{
                type: Input
            }], top: [{
                type: Input
            }], bottom: [{
                type: Input
            }], near: [{
                type: Input
            }], far: [{
                type: Input
            }] } });
