// ngx-three/esm2020/lib/generated/ThPass.mjs
/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { Pass } from 'three/examples/jsm/postprocessing/Pass';
import { ThPassBase } from '../ThPassBase';
import * as i0 from "@angular/core";
export class ThPass extends ThPassBase {
    getType() {
        return Pass;
    }
    set enabled(value) {
        if (this._objRef) {
            this._objRef.enabled = value;
        }
    }
    set needsSwap(value) {
        if (this._objRef) {
            this._objRef.needsSwap = value;
        }
    }
    set clear(value) {
        if (this._objRef) {
            this._objRef.clear = value;
        }
    }
    set renderToScreen(value) {
        if (this._objRef) {
            this._objRef.renderToScreen = value;
        }
    }
}
ThPass.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThPass, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThPass.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThPass, selector: "th-pass", inputs: { enabled: "enabled", needsSwap: "needsSwap", clear: "clear", renderToScreen: "renderToScreen" }, providers: [{ provide: ThPassBase, useExisting: forwardRef(() => ThPass) }], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThPass, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-pass',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThPassBase, useExisting: forwardRef(() => ThPass) }],
                }]
        }], propDecorators: { enabled: [{
                type: Input
            }], needsSwap: [{
                type: Input
            }], clear: [{
                type: Input
            }], renderToScreen: [{
                type: Input
            }] } });
