// ngx-three/esm2020/lib/generated/ThPointLight.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { PointLight } from 'three';
import { ThLight } from './ThLight';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThPointLight extends ThLight {
    getType() {
        return PointLight;
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
    set intensity(value) {
        if (this._objRef) {
            this._objRef.intensity = value;
        }
    }
    set distance(value) {
        if (this._objRef) {
            this._objRef.distance = value;
        }
    }
    set decay(value) {
        if (this._objRef) {
            this._objRef.decay = value;
        }
    }
    set shadow(value) {
        if (this._objRef) {
            this._objRef.shadow = value;
        }
    }
    set power(value) {
        if (this._objRef) {
            this._objRef.power = value;
        }
    }
}
ThPointLight.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThPointLight, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThPointLight.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThPointLight, selector: "th-pointLight", inputs: { type: "type", intensity: "intensity", distance: "distance", decay: "decay", shadow: "shadow", power: "power" }, providers: [
        { provide: ThObject3D, useExisting: forwardRef(() => ThPointLight) },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThPointLight, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-pointLight',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        { provide: ThObject3D, useExisting: forwardRef(() => ThPointLight) },
                    ],
                }]
        }], propDecorators: { type: [{
                type: Input
            }], intensity: [{
                type: Input
            }], distance: [{
                type: Input
            }], decay: [{
                type: Input
            }], shadow: [{
                type: Input
            }], power: [{
                type: Input
            }] } });
