// ngx-three/esm2020/lib/generated/ThRawShaderMaterial.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, } from '@angular/core';
import { RawShaderMaterial } from 'three';
import { ThMaterial } from './ThMaterial';
import { ThShaderMaterial } from './ThShaderMaterial';
import * as i0 from "@angular/core";
export class ThRawShaderMaterial extends ThShaderMaterial {
    getType() {
        return RawShaderMaterial;
    }
}
ThRawShaderMaterial.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThRawShaderMaterial, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThRawShaderMaterial.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThRawShaderMaterial, selector: "th-rawShaderMaterial", providers: [
        { provide: ThMaterial, useExisting: forwardRef(() => ThRawShaderMaterial) },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThRawShaderMaterial, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-rawShaderMaterial',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        { provide: ThMaterial, useExisting: forwardRef(() => ThRawShaderMaterial) },
                    ],
                }]
        }] });
