// ngx-three/esm2020/lib/generated/ThRectAreaLight.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { RectAreaLight } from 'three';
import { ThLight } from './ThLight';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThRectAreaLight extends ThLight {
    getType() {
        return RectAreaLight;
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
    set width(value) {
        if (this._objRef) {
            this._objRef.width = value;
        }
    }
    set height(value) {
        if (this._objRef) {
            this._objRef.height = value;
        }
    }
    set intensity(value) {
        if (this._objRef) {
            this._objRef.intensity = value;
        }
    }
    set power(value) {
        if (this._objRef) {
            this._objRef.power = value;
        }
    }
}
ThRectAreaLight.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThRectAreaLight, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThRectAreaLight.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThRectAreaLight, selector: "th-rectAreaLight", inputs: { type: "type", width: "width", height: "height", intensity: "intensity", power: "power" }, providers: [
        { provide: ThObject3D, useExisting: forwardRef(() => ThRectAreaLight) },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThRectAreaLight, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-rectAreaLight',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        { provide: ThObject3D, useExisting: forwardRef(() => ThRectAreaLight) },
                    ],
                }]
        }], propDecorators: { type: [{
                type: Input
            }], width: [{
                type: Input
            }], height: [{
                type: Input
            }], intensity: [{
                type: Input
            }], power: [{
                type: Input
            }] } });
