// ngx-three/esm2020/lib/generated/ThSSAARenderPass.mjs
/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input, } from '@angular/core';
import { SSAARenderPass } from 'three/examples/jsm/postprocessing/SSAARenderPass';
import { ThPassBase } from '../ThPassBase';
import { ThPass } from './ThPass';
import * as i0 from "@angular/core";
export class ThSSAARenderPass extends ThPass {
    getType() {
        return SSAARenderPass;
    }
    set scene(value) {
        if (this._objRef) {
            this._objRef.scene = value;
        }
    }
    set camera(value) {
        if (this._objRef) {
            this._objRef.camera = value;
        }
    }
    set sampleLevel(value) {
        if (this._objRef) {
            this._objRef.sampleLevel = value;
        }
    }
    set unbiased(value) {
        if (this._objRef) {
            this._objRef.unbiased = value;
        }
    }
    set clearColor(value) {
        if (this._objRef) {
            this._objRef.clearColor = value;
        }
    }
    set clearAlpha(value) {
        if (this._objRef) {
            this._objRef.clearAlpha = value;
        }
    }
    set copyUniforms(value) {
        if (this._objRef) {
            this._objRef.copyUniforms = value;
        }
    }
    set copyMaterial(value) {
        if (this._objRef) {
            this._objRef.copyMaterial = value;
        }
    }
    set fsQuad(value) {
        if (this._objRef) {
            this._objRef.fsQuad = value;
        }
    }
    set sampleRenderTarget(value) {
        if (this._objRef) {
            this._objRef.sampleRenderTarget = value;
        }
    }
}
ThSSAARenderPass.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThSSAARenderPass, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThSSAARenderPass.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThSSAARenderPass, selector: "th-sSAARenderPass", inputs: { scene: "scene", camera: "camera", sampleLevel: "sampleLevel", unbiased: "unbiased", clearColor: "clearColor", clearAlpha: "clearAlpha", copyUniforms: "copyUniforms", copyMaterial: "copyMaterial", fsQuad: "fsQuad", sampleRenderTarget: "sampleRenderTarget" }, providers: [
        { provide: ThPassBase, useExisting: forwardRef(() => ThSSAARenderPass) },
    ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThSSAARenderPass, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-sSAARenderPass',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        { provide: ThPassBase, useExisting: forwardRef(() => ThSSAARenderPass) },
                    ],
                }]
        }], propDecorators: { scene: [{
                type: Input
            }], camera: [{
                type: Input
            }], sampleLevel: [{
                type: Input
            }], unbiased: [{
                type: Input
            }], clearColor: [{
                type: Input
            }], clearAlpha: [{
                type: Input
            }], copyUniforms: [{
                type: Input
            }], copyMaterial: [{
                type: Input
            }], fsQuad: [{
                type: Input
            }], sampleRenderTarget: [{
                type: Input
            }] } });
