// @angular-three/core/esm2020/lib/stores/animation-frame.store.mjs
var _NgtAnimationFrameStore_instances, _NgtAnimationFrameStore_register, _NgtAnimationFrameStore_unregister;
import { __classPrivateFieldGet } from "tslib";
import { Injectable } from '@angular/core';
import { noop } from 'rxjs';
import { makeId } from '../utils/make-id';
import { EnhancedRxState, getActions } from './enhanced-rx-state';
import * as i0 from "@angular/core";
export class NgtAnimationFrameStore extends EnhancedRxState {
    constructor() {
        super();
        _NgtAnimationFrameStore_instances.add(this);
        this.actions = getActions();
        this.set({ animations: {}, subscribers: [], hasPriority: false });
        this.hold(this.select('animations'), (animations) => {
            const subscribers = Object.values(animations);
            subscribers.sort((a, b) => (a.priority || 0) - (b.priority || 0));
            const hasPriority = subscribers.some(({ priority }) => !!priority);
            this.set({ hasPriority, subscribers });
        });
        this.hold(this.actions.animationRecord$, __classPrivateFieldGet(this, _NgtAnimationFrameStore_instances, "m", _NgtAnimationFrameStore_register).bind(this));
        this.hold(this.actions.unsubscriberUuid$, __classPrivateFieldGet(this, _NgtAnimationFrameStore_instances, "m", _NgtAnimationFrameStore_unregister).bind(this));
    }
    register(animationRecord) {
        const uuid = animationRecord.obj?.uuid || makeId();
        this.actions.animationRecord({ ...animationRecord, uuid });
        return uuid;
    }
    ngOnDestroy() {
        this.set({ animations: {} });
        super.ngOnDestroy();
    }
}
_NgtAnimationFrameStore_instances = new WeakSet(), _NgtAnimationFrameStore_register = function _NgtAnimationFrameStore_register({ uuid, ...animationRecord } = {
    uuid: '',
    callback: noop,
}) {
    if (uuid) {
        this.set((state) => {
            return {
                animations: { ...state.animations, [uuid]: animationRecord },
            };
        });
    }
    return (prevRecord, isUnsub) => {
        if ((prevRecord && prevRecord.uuid !== uuid) || isUnsub) {
            __classPrivateFieldGet(this, _NgtAnimationFrameStore_instances, "m", _NgtAnimationFrameStore_unregister).call(this, uuid);
        }
    };
}, _NgtAnimationFrameStore_unregister = function _NgtAnimationFrameStore_unregister(uuid) {
    if (!uuid)
        return;
    this.set((state) => {
        const { [uuid]: _, ...animations } = state.animations;
        return { animations };
    });
};
NgtAnimationFrameStore.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtAnimationFrameStore, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
NgtAnimationFrameStore.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtAnimationFrameStore });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtAnimationFrameStore, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return []; } });
