// @angular-three/core/esm2020/lib/stores/enhanced-rx-state.mjs
import { Directive } from '@angular/core';
import { createSideEffectObservable, RxState } from '@rx-angular/state';
import { catchError, EMPTY, noop, Subject, tap } from 'rxjs';
import * as i0 from "@angular/core";
export class EnhancedRxState extends RxState {
    constructor() {
        super(...arguments);
        this.effect$ = createSideEffectObservable();
        this.effectSubscription = this.effect$.subscribe();
    }
    ngOnDestroy() {
        super.ngOnDestroy();
        this.effectSubscription.unsubscribe();
    }
    holdEffect(obsOrObsWithSideEffect, sideEffectFn) {
        let cleanupFn = noop;
        let firstRun = false;
        let latestValue = undefined;
        const sideEffect = obsOrObsWithSideEffect.pipe(catchError((_) => EMPTY));
        this.effect$.nextEffectObservable(sideEffect.pipe(tap({
            next: (value) => {
                if (cleanupFn && firstRun) {
                    cleanupFn(latestValue, false);
                }
                const cleanUpOrVoid = sideEffectFn(value, firstRun);
                if (cleanUpOrVoid) {
                    cleanupFn = cleanUpOrVoid;
                }
                latestValue = value;
                if (!firstRun) {
                    firstRun = true;
                }
            },
            unsubscribe: () => {
                if (cleanupFn) {
                    cleanupFn(latestValue, true);
                }
            },
        })));
    }
}
EnhancedRxState.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: EnhancedRxState, deps: null, target: i0.ɵɵFactoryTarget.Directive });
EnhancedRxState.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: EnhancedRxState, usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: EnhancedRxState, decorators: [{
            type: Directive
        }] });
export function capitalize(str) {
    return (str.charAt(0).toUpperCase() + str.slice(1));
}
/**
 * Returns a object based off of the provided typing with a separate setter `[prop](value: T[K]): void` and observable stream `[prop]$: Observable<T[K]>`;
 *
 * { search: string } => { search$: Observable<string>, search: (value: string) => void;}
 *
 * @example
 *
 * const actions = getActions<search: string, submit: void>({search: (e) => e.target.value});
 *
 * actions.search($event);
 * actions.search$ | async;
 *
 * @param transforms map of transform functions to apply on certain properties if they are set.
 */
export function getActions() {
    const subjects = {};
    const handler = {
        get(_, property) {
            if (property.toString().split('').pop() === '$') {
                const propName = property
                    .toString()
                    .slice(0, -1);
                subjects[propName] = subjects[propName] || new Subject();
                return subjects[propName];
            }
            return (args) => {
                subjects[property] = subjects[property] || new Subject();
                subjects[property].next(args);
            };
        },
        set() {
            throw new Error('No setters available. To emit call the property name.');
        },
    };
    return new Proxy({}, handler);
}
