// @angular-three/core/esm2020/lights/lib/point-light/point-light.directive.mjs
// GENERATED
import { NgtLight, NGT_OBJECT_CONTROLLER_PROVIDER } from '@angular-three/core';
import { NgModule, Directive, Input } from '@angular/core';
import * as THREE from 'three';
import * as i0 from "@angular/core";
export class NgtPointLight extends NgtLight {
    constructor() {
        super(...arguments);
        this.lightType = THREE.PointLight;
    }
    set args(v) {
        this.lightArgs = v;
    }
}
NgtPointLight.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtPointLight, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtPointLight.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtPointLight, selector: "ngt-point-light", inputs: { args: "args" }, providers: [
        {
            provide: NgtLight,
            useExisting: NgtPointLight,
        },
        NGT_OBJECT_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtPointLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtPointLight, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-point-light',
                    exportAs: 'ngtPointLight',
                    providers: [
                        {
                            provide: NgtLight,
                            useExisting: NgtPointLight,
                        },
                        NGT_OBJECT_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
export class NgtPointLightModule {
}
NgtPointLightModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtPointLightModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtPointLightModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtPointLightModule, declarations: [NgtPointLight], exports: [NgtPointLight] });
NgtPointLightModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtPointLightModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtPointLightModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtPointLight],
                    exports: [NgtPointLight],
                }]
        }] });
