// ngx-three/esm2020/lib/loaders/ThLoader.mjs
import { Directive, EventEmitter, Host, Input, Output } from '@angular/core';
import { createLazyObject3DProxy } from './LazyObject3dProxy';
import * as i0 from "@angular/core";
import * as i1 from "../generated/ThObject3D";
// eslint-disable-next-line @angular-eslint/directive-class-suffix
export class ThLoader {
    constructor(host, zone) {
        this.host = host;
        this.zone = zone;
        this.initialized = false;
        this.proxy = createLazyObject3DProxy();
        host.objRef = this.proxy;
    }
    set url(url) {
        this._url = url;
        this.loadAsync();
    }
    get url() {
        return this._url;
    }
    ngOnInit() {
        this.initialized = true;
        this.loadAsync();
        // this.zone.runOutsideAngular(() => );
    }
    get onLoaded() {
        if (!this.onLoaded$) {
            this.onLoaded$ = new EventEmitter();
        }
        return this.onLoaded$;
    }
    get onProgress() {
        if (!this.onProgress$) {
            this.onProgress$ = new EventEmitter();
        }
        return this.onProgress$;
    }
    async loadAsync() {
        if (!this.initialized) {
            return;
        }
        const loaderFn = this.loaderFn;
        if (!loaderFn) {
            throw new Error('Missing loader Function ( @Input() loaderFn )');
        }
        if (!this.proxy) {
            return;
        }
        const onProgress = this.onProgress$
            ? (progress) => {
                this.onProgress$?.next(progress);
            }
            : undefined;
        let loaderResult;
        const onLoaded = this.onLoaded$
            ? (result) => {
                loaderResult = result;
            }
            : undefined;
        const object = await this.zone.runOutsideAngular(() => loaderFn(this.url, onProgress, onLoaded));
        this.proxy.objRef = object;
        // add the new object to the parent and
        // emit a loaded event directly on the three.js object and on objRef$
        this.host.objRef = this.proxy;
        if (this.onLoaded$ && loaderResult !== undefined) {
            this.onLoaded$?.next(loaderResult);
        }
        return object;
    }
}
ThLoader.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThLoader, deps: [{ token: i1.ThObject3D, host: true }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive });
ThLoader.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.2", type: ThLoader, selector: "[load]", inputs: { loaderFn: "loaderFn", url: "url" }, outputs: { onLoaded: "onLoaded", onProgress: "onProgress" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThLoader, decorators: [{
            type: Directive,
            args: [{
                    selector: '[load]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThObject3D, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }]; }, propDecorators: { loaderFn: [{
                type: Input
            }], url: [{
                type: Input
            }], onLoaded: [{
                type: Output
            }], onProgress: [{
                type: Output
            }] } });
