// ngx-three/esm2020/lib/ThCanvas.mjs
import { ChangeDetectionStrategy, Component, ContentChildren, forwardRef, Inject, Input, ViewChild } from '@angular/core';
import { RaycasterService } from './events/raycaster.service';
import { ThObject3D } from './generated/ThObject3D';
import { ThEngineService } from './ThEngine.service';
import { ThView } from './ThView';
import * as i0 from "@angular/core";
import * as i1 from "./ThEngine.service";
import * as i2 from "./events/raycaster.service";
// eslint-disable-next-line @angular-eslint/component-class-suffix
export class ThCanvas extends ThView {
    constructor(engServ, raycaster) {
        super(engServ, raycaster);
        this.engServ = engServ;
        this.raycaster = raycaster;
        this.nid = ThCanvas.instanceCnt++;
        /**
         * if true does not use the ThCanvas as view
         * this is esepcially usefull if you have multiple views
         * and want to iterate over all of them with *ngFor
         * use statically (before ngOnInit is triggered)
         */
        this.disableDefaultView = false;
        console.log('canvas ' + this.nid);
    }
    get obj() {
        return this;
    }
    set views(viewList) {
        viewList.forEach((v) => this.engServ.addView(v));
    }
    set rendererCanvas(canvas) {
        if (!canvas) {
            return;
        }
        this._rendererCanvas = canvas;
        canvas.nativeElement.id += this.nid;
        this.engServ.setCanvas(canvas.nativeElement);
    }
    get rendererCanvas() {
        return this._rendererCanvas;
    }
    ngAfterViewInit() {
        this.engServ.requestAnimationFrame();
    }
    ngOnInit() {
        if (!this.rendererCanvas) {
            throw new Error('Missing Canvas');
        }
        this.engServ.setCanvas(this.rendererCanvas.nativeElement);
        super.ngOnInit();
        if (!this.disableDefaultView) {
            this.engServ.addView(this);
        }
    }
}
ThCanvas.instanceCnt = 0;
ThCanvas.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThCanvas, deps: [{ token: i1.ThEngineService }, { token: forwardRef(() => RaycasterService) }], target: i0.ɵɵFactoryTarget.Component });
ThCanvas.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThCanvas, selector: "th-canvas", inputs: { disableDefaultView: "disableDefaultView" }, providers: [
        { provide: ThObject3D, useExisting: forwardRef(() => ThCanvas) },
        ThEngineService,
        forwardRef(() => RaycasterService),
        { provide: ThView, useExisting: forwardRef(() => ThCanvas) }
    ], queries: [{ propertyName: "views", predicate: ThView }], viewQueries: [{ propertyName: "rendererCanvas", first: true, predicate: ["rendererCanvas"], descendants: true, static: true }], usesInheritance: true, ngImport: i0, template: '<canvas #rendererCanvas id="rendererCanvas" style="width: 100%; height: 100%"></canvas>', isInline: true, styles: [":host{display:flex;flex:1;height:100%;line-height:0}canvas{flex:1;outline:none}\n"], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThCanvas, decorators: [{
            type: Component,
            args: [{ selector: 'th-canvas', template: '<canvas #rendererCanvas id="rendererCanvas" style="width: 100%; height: 100%"></canvas>', changeDetection: ChangeDetectionStrategy.OnPush, providers: [
                        { provide: ThObject3D, useExisting: forwardRef(() => ThCanvas) },
                        ThEngineService,
                        forwardRef(() => RaycasterService),
                        { provide: ThView, useExisting: forwardRef(() => ThCanvas) }
                    ], styles: [":host{display:flex;flex:1;height:100%;line-height:0}canvas{flex:1;outline:none}\n"] }]
        }], ctorParameters: function () { return [{ type: i1.ThEngineService }, { type: i2.RaycasterService, decorators: [{
                    type: Inject,
                    args: [forwardRef(() => RaycasterService)]
                }] }]; }, propDecorators: { disableDefaultView: [{
                type: Input
            }], views: [{
                type: ContentChildren,
                args: [ThView]
            }], rendererCanvas: [{
                type: ViewChild,
                args: ['rendererCanvas', { static: true }]
            }] } });
