// ngx-three/esm2020/lib/ThControlBase.mjs
import { Component } from '@angular/core';
import { ThWrapperBase } from './ThWrapperBase';
import * as i0 from "@angular/core";
import * as i1 from "./generated/ThObject3D";
import * as i2 from "./ThCanvas";
// eslint-disable-next-line @angular-eslint/component-class-suffix
export class ThControlBase extends ThWrapperBase {
    constructor(camera, canvas) {
        super();
        this.camera = camera;
        this.canvas = canvas;
    }
    createThreeInstance(args) {
        if (!args) {
            args = [this.camera.objRef, this.canvas?.rendererCanvas?.nativeElement];
        }
        return super.createThreeInstance(args);
    }
}
ThControlBase.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThControlBase, deps: [{ token: i1.ThObject3D }, { token: i2.ThCanvas }], target: i0.ɵɵFactoryTarget.Component });
ThControlBase.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThControlBase, selector: "th-abs-control", usesInheritance: true, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThControlBase, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-abs-control',
                    template: ''
                }]
        }], ctorParameters: function () { return [{ type: i1.ThObject3D }, { type: i2.ThCanvas }]; } });
