// ngx-three/esm2020/lib/ThMaterialBase.mjs
import { Component } from '@angular/core';
import { ThWrapperBase } from './ThWrapperBase';
import * as i0 from "@angular/core";
import * as i1 from "./generated/ThObject3D";
// eslint-disable-next-line @angular-eslint/component-class-suffix
export class ThMaterialBase extends ThWrapperBase {
    constructor(parent) {
        super();
        this.parent = parent;
    }
    addToParent() {
        if (!this.parent.objRef) {
            throw new Error("parent object doesn't hold a three js object instance");
        }
        this.parent.objRef.material = this.objRef;
    }
    removeFromParent() {
        if (this._objRef && this.parent && this.parent.objRef) {
            this.parent.objRef.material = undefined;
        }
    }
}
ThMaterialBase.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThMaterialBase, deps: [{ token: i1.ThObject3D }], target: i0.ɵɵFactoryTarget.Component });
ThMaterialBase.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThMaterialBase, selector: "th-abs-material", usesInheritance: true, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThMaterialBase, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-abs-material',
                    template: ''
                }]
        }], ctorParameters: function () { return [{ type: i1.ThObject3D }]; } });
