// ngx-three/esm2020/lib/ThPassBase.mjs
import { Component, Optional } from '@angular/core';
import { ThWrapperBase } from './ThWrapperBase';
import * as i0 from "@angular/core";
import * as i1 from "./generated/overrides/ThEffectComposer";
// eslint-disable-next-line @angular-eslint/component-class-suffix
export class ThPassBase extends ThWrapperBase {
    constructor(effectComposer) {
        super();
        this.effectComposer = effectComposer;
    }
    addToParent() {
        if (this._objRef && this.effectComposer && this.effectComposer.objRef) {
            this.effectComposer.objRef.addPass(this._objRef);
        }
    }
    removeFromParent() {
        if (this._objRef && this.effectComposer && this.effectComposer.objRef) {
            this.effectComposer.objRef.removePass(this._objRef);
        }
    }
}
ThPassBase.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThPassBase, deps: [{ token: i1.ThEffectComposer, optional: true }], target: i0.ɵɵFactoryTarget.Component });
ThPassBase.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.2", type: ThPassBase, selector: "th-abs-control", usesInheritance: true, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.2", ngImport: i0, type: ThPassBase, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-abs-control',
                    template: ''
                }]
        }], ctorParameters: function () { return [{ type: i1.ThEffectComposer, decorators: [{
                    type: Optional
                }] }]; } });
