// @angular-three/core/fesm2020/angular-three-core-cameras.mjs
import * as i1 from '@angular-three/core';
import { NgtCommonCamera, NGT_OBJECT_CONTROLLER_PROVIDER, NGT_OBJECT_WATCHED_CONTROLLER } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, Input, NgModule, Inject } from '@angular/core';
import * as THREE from 'three';
import { __classPrivateFieldSet, __classPrivateFieldGet } from 'tslib';

// GENERATED
class NgtCamera extends NgtCommonCamera {
    constructor() {
        super(...arguments);
        this.cameraType = THREE.Camera;
    }
    set args(v) {
        this.cameraArgs = v;
    }
}
NgtCamera.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCamera, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtCamera.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtCamera, selector: "ngt-camera", inputs: { args: "args" }, providers: [
        {
            provide: NgtCommonCamera,
            useExisting: NgtCamera,
        },
        NGT_OBJECT_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCamera, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-camera',
                    exportAs: 'ngtCamera',
                    providers: [
                        {
                            provide: NgtCommonCamera,
                            useExisting: NgtCamera,
                        },
                        NGT_OBJECT_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
class NgtCameraModule {
}
NgtCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCameraModule, declarations: [NgtCamera], exports: [NgtCamera] });
NgtCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtCamera],
                    exports: [NgtCamera],
                }]
        }] });

// GENERATED
class NgtPerspectiveCamera extends NgtCommonCamera {
    constructor() {
        super(...arguments);
        this.cameraType = THREE.PerspectiveCamera;
    }
    set args(v) {
        this.cameraArgs = v;
    }
}
NgtPerspectiveCamera.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtPerspectiveCamera, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtPerspectiveCamera.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtPerspectiveCamera, selector: "ngt-perspective-camera", inputs: { args: "args" }, providers: [
        {
            provide: NgtCommonCamera,
            useExisting: NgtPerspectiveCamera,
        },
        NGT_OBJECT_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtPerspectiveCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtPerspectiveCamera, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-perspective-camera',
                    exportAs: 'ngtPerspectiveCamera',
                    providers: [
                        {
                            provide: NgtCommonCamera,
                            useExisting: NgtPerspectiveCamera,
                        },
                        NGT_OBJECT_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
class NgtPerspectiveCameraModule {
}
NgtPerspectiveCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtPerspectiveCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtPerspectiveCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtPerspectiveCameraModule, declarations: [NgtPerspectiveCamera], exports: [NgtPerspectiveCamera] });
NgtPerspectiveCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtPerspectiveCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtPerspectiveCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtPerspectiveCamera],
                    exports: [NgtPerspectiveCamera],
                }]
        }] });

// GENERATED
class NgtOrthographicCamera extends NgtCommonCamera {
    constructor() {
        super(...arguments);
        this.cameraType = THREE.OrthographicCamera;
    }
    set args(v) {
        this.cameraArgs = v;
    }
}
NgtOrthographicCamera.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtOrthographicCamera, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtOrthographicCamera.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtOrthographicCamera, selector: "ngt-orthographic-camera", inputs: { args: "args" }, providers: [
        {
            provide: NgtCommonCamera,
            useExisting: NgtOrthographicCamera,
        },
        NGT_OBJECT_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtOrthographicCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtOrthographicCamera, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-orthographic-camera',
                    exportAs: 'ngtOrthographicCamera',
                    providers: [
                        {
                            provide: NgtCommonCamera,
                            useExisting: NgtOrthographicCamera,
                        },
                        NGT_OBJECT_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
class NgtOrthographicCameraModule {
}
NgtOrthographicCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtOrthographicCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtOrthographicCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtOrthographicCameraModule, declarations: [NgtOrthographicCamera], exports: [NgtOrthographicCamera] });
NgtOrthographicCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtOrthographicCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtOrthographicCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtOrthographicCamera],
                    exports: [NgtOrthographicCamera],
                }]
        }] });

// GENERATED
class NgtArrayCamera extends NgtCommonCamera {
    constructor() {
        super(...arguments);
        this.cameraType = THREE.ArrayCamera;
    }
    set args(v) {
        this.cameraArgs = v;
    }
}
NgtArrayCamera.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtArrayCamera, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtArrayCamera.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtArrayCamera, selector: "ngt-array-camera", inputs: { args: "args" }, providers: [
        {
            provide: NgtCommonCamera,
            useExisting: NgtArrayCamera,
        },
        NGT_OBJECT_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtArrayCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtArrayCamera, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-array-camera',
                    exportAs: 'ngtArrayCamera',
                    providers: [
                        {
                            provide: NgtCommonCamera,
                            useExisting: NgtArrayCamera,
                        },
                        NGT_OBJECT_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
class NgtArrayCameraModule {
}
NgtArrayCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtArrayCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtArrayCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtArrayCameraModule, declarations: [NgtArrayCamera], exports: [NgtArrayCamera] });
NgtArrayCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtArrayCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtArrayCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtArrayCamera],
                    exports: [NgtArrayCamera],
                }]
        }] });

// GENERATED
class NgtStereoCamera extends NgtCommonCamera {
    constructor() {
        super(...arguments);
        this.cameraType = THREE.StereoCamera;
    }
    set args(v) {
        this.cameraArgs = v;
    }
}
NgtStereoCamera.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtStereoCamera, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtStereoCamera.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtStereoCamera, selector: "ngt-stereo-camera", inputs: { args: "args" }, providers: [
        {
            provide: NgtCommonCamera,
            useExisting: NgtStereoCamera,
        },
        NGT_OBJECT_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtStereoCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtStereoCamera, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-stereo-camera',
                    exportAs: 'ngtStereoCamera',
                    providers: [
                        {
                            provide: NgtCommonCamera,
                            useExisting: NgtStereoCamera,
                        },
                        NGT_OBJECT_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
class NgtStereoCameraModule {
}
NgtStereoCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtStereoCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtStereoCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtStereoCameraModule, declarations: [NgtStereoCamera], exports: [NgtStereoCamera] });
NgtStereoCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtStereoCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtStereoCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtStereoCamera],
                    exports: [NgtStereoCamera],
                }]
        }] });

var _NgtCubeCamera_cubeCamera;
class NgtCubeCamera {
    constructor(objectController, ngZone) {
        this.objectController = objectController;
        this.ngZone = ngZone;
        _NgtCubeCamera_cubeCamera.set(this, void 0);
        objectController.initFn = () => {
            __classPrivateFieldSet(this, _NgtCubeCamera_cubeCamera, new THREE.CubeCamera(...this.args), "f");
            return __classPrivateFieldGet(this, _NgtCubeCamera_cubeCamera, "f");
        };
    }
    get cubeCamera() {
        return __classPrivateFieldGet(this, _NgtCubeCamera_cubeCamera, "f");
    }
    ngOnInit() {
        this.ngZone.runOutsideAngular(() => {
            this.objectController.init();
        });
    }
}
_NgtCubeCamera_cubeCamera = new WeakMap();
NgtCubeCamera.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCubeCamera, deps: [{ token: NGT_OBJECT_WATCHED_CONTROLLER }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive });
NgtCubeCamera.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtCubeCamera, selector: "ngt-cube-camera", inputs: { args: "args" }, providers: [NGT_OBJECT_CONTROLLER_PROVIDER], exportAs: ["ngtCubeCamera"], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCubeCamera, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-cube-camera',
                    exportAs: 'ngtCubeCamera',
                    providers: [NGT_OBJECT_CONTROLLER_PROVIDER],
                }]
        }], ctorParameters: function () { return [{ type: i1.NgtObject3dController, decorators: [{
                    type: Inject,
                    args: [NGT_OBJECT_WATCHED_CONTROLLER]
                }] }, { type: i0.NgZone }]; }, propDecorators: { args: [{
                type: Input
            }] } });
class NgtCubeCameraModule {
}
NgtCubeCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCubeCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtCubeCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCubeCameraModule, declarations: [NgtCubeCamera], exports: [NgtCubeCamera] });
NgtCubeCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCubeCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtCubeCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtCubeCamera],
                    exports: [NgtCubeCamera],
                }]
        }] });

// GENERATED

/**
 * Generated bundle index. Do not edit.
 */

export { NgtArrayCamera, NgtArrayCameraModule, NgtCamera, NgtCameraModule, NgtCubeCamera, NgtCubeCameraModule, NgtOrthographicCamera, NgtOrthographicCameraModule, NgtPerspectiveCamera, NgtPerspectiveCameraModule, NgtStereoCamera, NgtStereoCameraModule };
