// @angular-three/core/fesm2020/angular-three-core-sprites.mjs
import { NgtCommonSprite, NGT_OBJECT_CONTROLLER_PROVIDER } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, NgModule } from '@angular/core';
import * as THREE from 'three';

// GENERATED
class NgtSprite extends NgtCommonSprite {
    constructor() {
        super(...arguments);
        this.spriteType = THREE.Sprite;
    }
}
NgtSprite.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtSprite, deps: null, target: i0.ɵɵFactoryTarget.Directive });
NgtSprite.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.0.3", type: NgtSprite, selector: "ngt-sprite", providers: [
        {
            provide: NgtCommonSprite,
            useExisting: NgtSprite,
        },
        NGT_OBJECT_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtSprite"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtSprite, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-sprite',
                    exportAs: 'ngtSprite',
                    providers: [
                        {
                            provide: NgtCommonSprite,
                            useExisting: NgtSprite,
                        },
                        NGT_OBJECT_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });
class NgtSpriteModule {
}
NgtSpriteModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtSpriteModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgtSpriteModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtSpriteModule, declarations: [NgtSprite], exports: [NgtSprite] });
NgtSpriteModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtSpriteModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: NgtSpriteModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgtSprite],
                    exports: [NgtSprite],
                }]
        }] });

// GENERATED

/**
 * Generated bundle index. Do not edit.
 */

export { NgtSprite, NgtSpriteModule };
